src/privatesend/privatesend.cpp

```diff
@@ -115,10 +116,11 @@ bool CPrivateSendBroadcastTx::CheckSignature(const CBLSPublicKey& blsPubKey) con
     return true;
 }
 
-bool CPrivateSendBroadcastTx::IsExpired(int nHeight)
+bool CPrivateSendBroadcastTx::IsExpired(const CBlockIndex* pindex)
 {
-    // expire confirmed DSTXes after ~1h since confirmation
-    return (nConfirmedHeight != -1) && (nHeight - nConfirmedHeight > 24);
+    // expire confirmed DSTXes after ~1h since confirmation or chainlocked confirmation
+    return (nConfirmedHeight != -1) && ((pindex->nHeight - nConfirmedHeight > 24) ||
```

Comment:
Hmm, can you replace this whole return statement with multiple ifs, like this:
```
if (nConfirmedHeight == -1 || pindex->nHeight < nConfirmedHeight) return false; // not mined yet
if (pindex->nHeight - nConfirmedHeight > 24) return true; // mined more then an hour ago
return llmq::chainLocksHandler->HasChainLock(pindex->nHeight, *pindex->phashBlock);
```

Reply:
IMHO these long return statements with multiple `&&` and `||` are always extremely hard to read, especially when each part actually checks a different thing. Using multiple ifs and a final return statement makes the intent much more clear.

Reply:
resolved, I agree